packages/workspace/src/generators/move/lib/normalize-schema.ts

```diff
@@ -119,84 +128,75 @@ function validateName(
 
 function getProjectNameAndRootFormats(
   tree: Tree,
-  schema: Schema,
+  options: Schema,
   projectConfiguration: ProjectConfiguration
 ): ProjectNameAndRootFormats {
-  let destination = normalizePathSlashes(schema.destination);
-  const normalizedNewProjectName = schema.newProjectName
-    ? names(schema.newProjectName).fileName
-    : undefined;
+  let destination = normalizePathSlashes(options.destination);
 
-  const asProvidedProjectName = normalizedNewProjectName ?? schema.projectName;
-  const asProvidedDestination = destination;
+  if (
+    options.newProjectName &&
+    options.newProjectName.includes('/') &&
+    !options.newProjectName.startsWith('@')
+  ) {
+    throw new Error(
+      `You can't specify a new project name with a directory path (${options.newProjectName}). ` +
+        `Please provide a valid name without path segments and the full destination with the "--destination" option.`
+    );
+  }
+
+  const asProvidedOptions = getAsProvidedOptions(
+    tree,
+    { ...options, destination },
+    projectConfiguration
+  );
 
-  if (normalizedNewProjectName?.startsWith('@')) {
+  if (options.projectNameAndRootFormat === 'as-provided') {
     return {
-      'as-provided': {
-        destination: asProvidedDestination,
-        importPath:
-          schema.importPath ??
-          // keep the existing import path if the name didn't change
-          (normalizedNewProjectName &&
-          schema.projectName !== normalizedNewProjectName
-            ? asProvidedProjectName
-            : undefined),
-        newProjectName: asProvidedProjectName,
-      },
+      'as-provided': asProvidedOptions,
+      derived: undefined,
```

Comment:
If derived is undefined, what will the prompt show for the user?

Reply:
It won't prompt and `as-provided` will be used.

This particular return statement happens when the `projectNameAndRootFormat` is `as-provided`. The other one below would have a warning explaining why it's using `as-provided` without prompting.